integration-tests/src/tests/governance.rs

```diff
@@ -0,0 +1,475 @@
+use pallet_vesting::VestingInfo;
+use polimec_base_runtime::{
+	Balances, Council, Democracy, Elections, ParachainStaking, Preimage, RuntimeOrigin, TechnicalCommittee, Treasury,
+	Vesting,
+};
+use tests::defaults::*;
+use xcm_emulator::get_account_id_from_seed;
+generate_accounts!(PEPE, CARLOS,);
+
+/// Test that an account with vested tokens (a lock) can use those tokens for a hold.
+/// The hold can also be released or slashed while the lock is still in place.
+#[test]
+fn vested_tokens_and_holds_work_together() {
+	PolimecBase::execute_with(|| {
+		let alice = PolimecBase::account_id_of(ALICE);
+		let new_account = create_vested_account();
+
+		assert_eq!(Balances::balance(&alice), 220 * PLMC - ED);
+		assert_eq!(Balances::balance(&new_account), 200 * PLMC + ED);
+
+		assert_ok!(Balances::hold(
+			&polimec_base_runtime::RuntimeHoldReason::ParachainStaking(
+				pallet_parachain_staking::HoldReason::StakingCollator
+			),
+			&new_account,
+			200 * PLMC
+		));
+		Balances::set_lock(*b"plmc/gov", &new_account, 200 * PLMC + ED, WithdrawReasons::all());
+		assert_ok!(Balances::release(
+			&polimec_base_runtime::RuntimeHoldReason::ParachainStaking(
+				pallet_parachain_staking::HoldReason::StakingCollator
+			),
+			&new_account,
+			200 * PLMC,
+			Precision::Exact
+		));
+
+		assert_eq!(Balances::reserved_balance(&new_account), 0);
+
+		assert_ok!(Balances::hold(
+			&polimec_base_runtime::RuntimeHoldReason::ParachainStaking(
+				pallet_parachain_staking::HoldReason::StakingCollator
+			),
+			&new_account,
+			200 * PLMC
+		));
+		let slashed = Balances::slash(
+			&polimec_base_runtime::RuntimeHoldReason::ParachainStaking(
+				pallet_parachain_staking::HoldReason::StakingCollator,
+			),
+			&new_account,
+			200 * PLMC,
+		);
+		assert_eq!(slashed.0.peek(), 200 * PLMC);
+
+		assert_eq!(Balances::reserved_balance(&new_account), 0);
+	})
+}
+
+/// Test that an account with vested tokens (a lock) cannot use those tokens for a reserve.
+#[test]
+fn vested_tokens_and_reserves_dont_work_together() {
+	PolimecBase::execute_with(|| {
+		let alice = PolimecBase::account_id_of(ALICE);
+		let new_account = create_vested_account();
+
+		assert_eq!(Balances::balance(&alice), 220 * PLMC - ED);
+		assert_eq!(Balances::balance(&new_account), 200 * PLMC + ED);
+
+		assert_noop!(
+			Balances::reserve(&new_account, 200 * PLMC),
+			pallet_balances::Error::<polimec_base_runtime::Runtime>::LiquidityRestrictions
+		);
+	});
+}
+
+/// Test that locks and freezes can be placed on balance that is already reserved.
+#[test]
+fn lock_and_freeze_after_reserve_does_work() {
+	PolimecBase::execute_with(|| {
+		let alice = PolimecBase::account_id_of(ALICE);
+
+		assert_ok!(Balances::reserve(&alice, 400 * PLMC));
+		assert_ok!(Balances::set_freeze(
+			&polimec_base_runtime::RuntimeFreezeReason::Democracy(pallet_democracy::FreezeReason::Vote),
+			&alice,
+			400 * PLMC
+		));
+		Balances::set_lock(*b"py/trsry", &alice, 400 * PLMC, WithdrawReasons::all());
```

Comment:
Its mostly that Reserves after locks do not work. So trying to use a reserve on your vesting scheduled tokens will result in an error. So for example people couldn't propose or second a public proposal with their vesting tokens or couldn't candidate themselves in election as it requires reserve. So thats mainly why we moved away from old traits for democracy and election pallets. Election for example also requires a reserve for voting in the original pallet, but we changed that to a vote locking period now. Another change that we made compared to the old pallets was being able to use total_balance instead of free balance for voting else tokens in staking couldn't be used for voting. But yeah you can place a lock/freeze on  reserved balance but you can't do it the other way around.